File: app/src/main/java/com/u1fukui/bbs/di/AppComponent.java
package com.u1fukui.bbs.di;

import com.u1fukui.bbs.App;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {AndroidSupportInjectionModule.class, AppModule.class, InjectorsModule.class})
public interface AppComponent extends AndroidInjector<App> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<App> {

        public abstract void appModule(AppModule activityModule);

        @Override
        public void seedInstance(App instance) {
            appModule(new AppModule(instance));
        }
    }
}

File: app/src/main/java/com/u1fukui/bbs/di/AppComponent.kt
package com.u1fukui.bbs.di

import com.u1fukui.bbs.App
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AndroidSupportInjectionModule::class, AppModule::class, InjectorsModule::class))
interface AppComponent : AndroidInjector<App> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<App>() {

        abstract fun appModule(activityModule: AppModule)

        override fun seedInstance(instance: App) {
            appModule(AppModule(instance))
        }
    }
}
